tests/test_diaCalculationPlugins.py

```diff
@@ -904,6 +912,162 @@ def testCalculate(self):
+
+        objId = 0
+        n_sources = 2
+        times, fluxes = self.GeneratePeriodicData(n_sources, period=10)
+        diaObjects = pd.DataFrame({"diaObjectId": [objId]})
+        diaSources = pd.DataFrame(
+            data={"diaObjectId": n_sources * [objId],
+                  "band": n_sources * ["u"],
+                  "diaSourceId": np.arange(n_sources, dtype=int),
+                  "midpointMjdTai": times,
+                  "psfFlux": fluxes,
+                  "psfFluxErr": 1e-3+np.zeros(n_sources)})
+
+        plugin = LombScarglePeriodogramMulti(LombScarglePeriodogramMultiConfig(),
+                                             "ap_lombScarglePeriodogramMulti",
+                                             None)
+
+        run_multi_plugin(diaObjects, diaSources, "u", plugin)
+        self.assertTrue(np.isnan(diaObjects.at[objId, "multiPeriod"]))
+        self.assertTrue(np.isnan(diaObjects.at[objId, "multiPower"]))
+        self.assertTrue(np.isnan(diaObjects.at[objId, "multiFap"]))
+
+
+class TestLombScarglePeriodogram(unittest.TestCase):
+
+    def GeneratePeriodicData(self, n=10, period=10):
+        """Generate periodic data for testing Lomb Scargle Periodogram.
+
+        Parameters
+        ----------
+        n : int
+            Number of data points.
+        period : float
+            Period of the periodic signal.
+
+        Returns
+        -------
+        t : np.ndarray
+            Time values.
+        y_obs : np.ndarray
+            Observed flux values.
+        """
+        np.random.seed(42)
+
+        # Generate random sparse time array.
+        t = np.linspace(-2*np.pi, 2*np.pi, n) + 100*np.random.random(n)
+        y = 10 + np.sin(2 * np.pi * t / period)
+        dy = 0.001 + 0.001 * np.random.random(n)
+        y_obs = np.random.normal(y, dy)
+
+        return t, y_obs
+
+    def testCalculate(self):
+        """Test Lomb Scargle Periodogram."""
+        n_sources = 10
+        objId = 0
+
+        # Test period calculation.
+        times, fluxes = self.GeneratePeriodicData(n_sources, period=10)
+        diaObjects = pd.DataFrame({"diaObjectId": [objId]})
+        diaSources = pd.DataFrame(
+            data={"diaObjectId": n_sources * [objId],
+                  "band": n_sources * ["u"],
+                  "diaSourceId": np.arange(n_sources, dtype=int),
+                  "midpointMjdTai": times,
+                  "psfFlux": fluxes,
+                  "psfFluxErr": 1e-3+np.zeros(n_sources)})
+
+        plugin = LombScarglePeriodogram(LombScarglePeriodogramConfig(),
+                                        "ap_lombScarglePeriodogram",
+                                        None)
+
+        run_multi_plugin(diaObjects, diaSources, "u", plugin)
+        # Period 10 days within rtol=0.05 atol=0.002
+        self.assertAlmostEqual(diaObjects.at[objId, "u_period"],
+                               10.03436005075541)
+
+        # Input nans to fluxes.
+        fluxes[4] = np.nan
```

Comment:
I don't understand the comment here. Is the goal of this to demonstrate that the code works if a flux is NaN? Maybe this? And I think you don't need to re-make diaObjects or diaSources, just change diaSources in-place.

```suggestion
        # Test that we get the same result with a NaN flux.
        diaSources["psfFlux"][4] = np.nan
```

Reply:
Yes, I wanted this code to demonstrate that it works with NaN present in the flux. I have address the comment.